src/commands/cli/releasenotes.ts

```diff
@@ -0,0 +1,180 @@
+      char: 'm',
+      default: false,
+    }),
+  };
+  private octokit!: Octokit;
+  private usernames: Record<string, string> = {};
+
+  public async run(): Promise<ChangesByPlugin> {
+    const auth = ensureString(new Env().getString('GH_TOKEN'), 'GH_TOKEN is required to be set in the environment');
+    this.octokit = new Octokit({ auth });
+    const cli = ensure<CLI>(this.flags.cli);
+    const fullName = cli === CLI.SF ? '@salesforce/cli' : 'sfdx-cli';
+    const npmPackage = this.getNpmPackage(fullName, this.flags.since ?? 'latest-rc');
+    const publishDate = npmPackage.time[npmPackage.version];
+    const plugins = this.normalizePlugins(npmPackage);
+    const changesByPlugin: ChangesByPlugin = {};
+    for (const plugin of plugins) {
+      const changes = await this.getPullsForPlugin(plugin, publishDate);
+      if (changes.length) changesByPlugin[plugin] = changes;
+    }
+
+    if (this.flags.markdown) {
+      this.logChangesMarkdown(changesByPlugin);
+    } else {
+      this.logChanges(changesByPlugin);
+    }
+
+    return changesByPlugin;
+  }
+
+  private getNpmPackage(name: string, version = 'latest'): NpmPackage {
+    const result = exec(`npm view ${name}@${version} --json`, { silent: true });
+    return JSON.parse(result.stdout) as NpmPackage;
+  }
+
+  private normalizePlugins(npmPackage: NpmPackage): string[] {
+    const plugins = npmPackage.oclif?.plugins ?? [];
+    const normalized = plugins
+      .filter((p) => !p.startsWith('@oclif'))
+      .map((p) => {
+        if (npmPackage.dependencies[p].startsWith('npm:')) {
+          return parseAliasedPackageName(npmPackage.dependencies[p]);
+        }
+        return p;
+      });
+    return [npmPackage.name, ...normalized];
+  }
+
+  private async getNameOfUser(username: string): Promise<string> {
+    if (this.usernames[username]) return this.usernames[username];
+
+    const { data } = await this.octokit.request('GET /users/{username}', { username });
+    const name = (data.name ?? data.login ?? username) as string;
+    this.usernames[username] = name;
+    return name;
+  }
+
+  private async getPullsForPlugin(plugin: string, since: string): Promise<Change[]> {
+    const npmPackage = this.getNpmPackage(plugin);
+    const homepage = npmPackage.homepage ?? (npmPackage.name === 'salesforce-alm' ? 'salesforcecli/toolbelt' : null);
+    if (!homepage) {
+      throw new SfdxError(`No github url found for ${npmPackage.name}`, 'GitUrlNotFound');
+    }
+    const [owner, repo] = homepage.replace('https://github.com/', '').replace(/#(.*)/g, '').split('/');
+    const pullRequests = await this.octokit.request('GET /repos/{owner}/{repo}/pulls', {
+      owner,
+      repo,
+      state: 'closed',
+      // eslint-disable-next-line camelcase
+      per_page: 100,
+    });
+    const changes = (await Promise.all(
+      pullRequests.data
+        .filter((pr) => {
+          return pr.merged_at && pr.merged_at > since && !pr.user.login.includes('dependabot');
```

Comment:
This won't work correctly if `since` is a semver because `pr.merged_at` is a date string. See the response example at [https://docs.github.com/en/rest/reference/pulls](https://docs.github.com/en/rest/reference/pulls#:~:text=26T19%3A01%3A12Z%22%2C-,%22merged_at%22%3A%20%222011-01-26T19%3A01%3A12Z%22%2C,-%22merge_commit_sha%22%3A%20%22e5bd3914e2e596debea16f433f57875b5b90bcd6%22%2C%0A%20%20%20%20%22assignee).

Reply:
@RodEsp `this.flags.since` is a version string (e.g. `7.124.0`) but `since` is the `publishDate`, which is instantiated here: https://github.com/salesforcecli/plugin-release-management/pull/338/files#diff-cbeedc973658dbd98d1b4b3a098413d7e9eb913a8540a8a990f7b354a1ca4f99R63

I'll update the variable to `publishDate` to avoid confusion